### ISSUE TITLE
Weight generated with T.shared_randomstreams.RandomStreams()

### ISSUE THREAD

**Issue by ad26kt** [Theano/Theano#4903]
partial code as follows:( please just pay attention to the context of the code)

rng = T.shared_randomstreams.RandomStreams()

class gen_rand(object):
    def __init__(self, rng, input):
        self.input_shape = input.shape 
        print type(self.input_shape)
        self.output = rng.uniform(size=self.input_shape, low=0, high=1)
    def return_output(self):
        return self.output

I coded a neural network code with one of the weigh W initialized with T.shared_randomstreams.RandomStreams(), the reason I didn't use numpy.random is that I don't want to feed input.shape everytime, but to compute the shape of input in the code.

the code works but just because it's random tensor, It can't be used as a Weight in neural network, it changes every time.

How can I initialize a weight in NN with random module in theano( just want to randomly generate value once at the beginning and not to be updated by itself, i tried 'no_default_updates=True' then it can't be updated through gradient descent!!, I also tried copy modue in python to shallow copy rng.uniform, there was an error. I tried numpy.random, but it requires numerical size of the random value but not tensor)

**Comment by lamblin** [comment]
Please use StackOverflow or theano-users for such questions.
In short, if you want an initial value for weights, you should use a shared variable initialized by values sampled by numpy.random. Usually, the size of the weight matrix does not change during training, so you should be able to know it in advance.
